datalad_next/constraints/basic.py

```diff
@@ -283,6 +283,39 @@ def short_description(self):
         return '%s:{%s}' % (self._key, ', '.join([repr(c) for c in self._allowed]))
 
 
+class EnsureSet(Constraint):
+    """Ensure an input contains all elements of a set of required values"""
+
+    def __init__(self, *values):
+        """
+        Parameters
+        ----------
+        *values
+           required values.
+        """
+        self._required = set(values)
+        super(EnsureSet, self).__init__()
+
+    def __call__(self, value):
+        if not self._required.issubset(set(value)):
+            self.raise_for(
+                value,
+                "missing {required}",
+                required=self._required,
+            )
+        return value
+
+    def long_description(self):
+        return 'All following fields are required: [CMD: %s CMD][PY: %s PY]' % (
+            str(tuple(i for i in self._required if i is not None)),
+            str(self._required)
```

Comment:
I see the point, and that `EnsureChoice` is doing the exact same thing. However, this is precisely what #280 is about. It is incomprehensible that this is because `None` cannot be given on the CLI, hence it should not be presented in the list of requirements (not sure if that is perfectly sound here, but for `EnsureChoice` this is what it is -- I made it such).

Just venting...

Reply:
Yes, I struggled with messaging...